diskfs.go

```diff
@@ -168,16 +180,24 @@ func writableMode(mode OpenModeOption) bool {
 	return false
 }
 
-func initDisk(f *os.File, openMode OpenModeOption) (*disk.Disk, error) {
+func initDisk(f *os.File, openMode OpenModeOption, sectorSize SectorSize) (*disk.Disk, error) {
 	var (
 		diskType      disk.Type
 		size          int64
-		lblksize      = int64(defaultBlocksize)
-		pblksize      = int64(defaultBlocksize)
+		lblksize      int64
+		pblksize      int64
 		defaultBlocks = true
 	)
 	log.Debug("initDisk(): start")
 
+	if sectorSize != SectorSizeDefault {
```

Comment:
This kind of if/else is a bit non-idiomatic in go. Normally you would keep the default setting, and override if needed. Sort of:

```go
var (
		lblksize      = int64(defaultBlocksize)
 		pblksize      = int64(defaultBlocksize)
)

// ...

// change it if it is different
if lblksize != sectorSize {
		lblksize = int64(sectorSize)
 		pblksize = int64(sectorSize)
}
```